### PR TITLE
Mocktest1204

### PR DESCRIPTION
**Pull request by snehalkotwal**

### CODE REVIEW

File: test/module/Entra/Add-EntraDirectoryRoleMember.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        }
+        It "Should contain OdataId in parameters when passed RefObjectId to it" {
+            Mock -CommandName New-MgDirectoryRoleMemberByRef -MockWith {$args} -ModuleName Microsoft.Graph.Entra
+
+            $result = Add-EntraDirectoryRoleMember -ObjectId "aaaaaaaa-0000-1111-2222-bbbbbbbbbbbb" -RefObjectId "bbbbbbbb-1111-2222-3333-cccccccccccc"
+            $params = Get-Parameters -data $result
+            $value="https://graph.microsoft.com/v1.0/directoryObjects/"
+            $params.OdataId | Should -Be $value"bbbbbbbb-1111-2222-3333-cccccccccccc"
+        }
+        It "Should contain 'User-Agent' header" {
```
Comment (KenitoInc) [inline test/module/Entra/Add-EntraDirectoryRoleMember.Tests.ps1:48]: This `User-Agent` test should be in this format
https://github.com/microsoftgraph/entra-powershell/blob/86e55e47c47a0cb99444a38bf6db950084df65cd/test/module/Entra/Get-EntraUser.Tests.ps1#L72-#L84

Check all the tests in this PR and ensure they are fixed

Comment (snehalkotwal) [reply test/module/Entra/Add-EntraDirectoryRoleMember.Tests.ps1:48]: Changes done.

File: test/module/Entra/New-EntraRoleAssignment.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        )
+    }
+
+    Mock -CommandName New-MgRoleManagementDirectoryRoleAssignment -MockWith $scriptblock -ModuleName Microsoft.Graph.Entra
+}
+
+Describe "New-EntraRoleAssignment" {
+Context "Test for New-EntraRoleAssignment" {
+        It "Should return created Ms role assignment" {
+            $result = New-EntraRoleAssignment -PrincipalId "aaaaaaaa-bbbb-cccc-1111-222222222222" -RoleDefinitionId  "54d418b2-4cc0-47ee-9b39-e8f84ed8e073" -DirectoryScopeId "/54d418b2-4cc0-47ee-9b39-e8f84ed8e073"
```
Comment (KenitoInc) [inline test/module/Entra/New-EntraRoleAssignment.Tests.ps1:35]: `New-EntraRoleAssignment` was renamed

File: test/module/Entra/Remove-EntraRoleAssignment.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    Import-Module (Join-Path $psscriptroot "..\Common-Functions.ps1") -Force
+
+    Mock -CommandName Remove-MgRoleManagementDirectoryRoleAssignment -MockWith {} -ModuleName Microsoft.Graph.Entra
+}
+  
+Describe "Remove-EntraRoleAssignment" {
+    Context "Test for Remove-EntraRoleAssignment" {
+        It "Should return empty object" {
+            $result = Remove-EntraRoleAssignment -Id "Aa1Bb2Cc3.-Dd4Ee5Ff6Gg7Hh8Ii9_~Jj0Kk1Ll2"
```
Comment (KenitoInc) [inline test/module/Entra/Remove-EntraRoleAssignment.Tests.ps1:16]: `Remove-EntraRoleAssignment`  was renamed

File: test/module/Entra/Remove-EntraRoleDefinition.Tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    Import-Module (Join-Path $psscriptroot "..\Common-Functions.ps1") -Force
+
+    Mock -CommandName Remove-MgRoleManagementDirectoryRoleDefinition -MockWith {} -ModuleName Microsoft.Graph.Entra
+}
+
+Describe "Remove-EntraRoleDefinition" {
+    Context "Test for Remove-EntraRoleDefinition" {
+        It "Should return empty object" {
+            $result = Remove-EntraRoleDefinition -Id "00aa00aa-bb11-cc22-dd33-44ee44ee44ee"
```
Comment (KenitoInc) [inline test/module/Entra/Remove-EntraRoleDefinition.Tests.ps1:16]: `Remove-EntraRoleDefinition` was renamed

File: test/module/Entra/Set-EntraRoleDefinition.Tests.ps1 [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+    Mock -CommandName Update-MgRoleManagementDirectoryRoleDefinition -MockWith {} -ModuleName Microsoft.Graph.Entra
+}
+  
+Describe "Set-EntraRoleDefinition" {
+    Context "Test for Set-EntraRoleDefinition" {
+        It "Should return empty object" {
+            $RolePermissions = New-object Microsoft.Open.MSGraph.Model.RolePermission
+            $RolePermissions.AllowedResourceActions =  @("microsoft.directory/applications/basic/read")
+            $result = Set-EntraRoleDefinition -Id "00aa00aa-bb11-cc22-dd33-44ee44ee44ee" -RolePermissions $RolePermissions -IsEnabled $false -DisplayName 'Mock-App' -ResourceScopes "/" -Description "Mock-App" -TemplateId "11bb11bb-cc22-dd33-ee44-55ff55ff55ff" -Version 3
```
Comment (KenitoInc) [inline test/module/Entra/Set-EntraRoleDefinition.Tests.ps1:18]: cmdlet renamed

File: test/module/Entra/Get-EntraDirectoryRoleAssignment.Tests.ps1 [hunk 6]
```diff
@@ -0,0 +1,123 @@
...
+              "AdditionalProperties"         = @{"@odata.context" = "https://graph.microsoft.com/v1.0/$metadata#roleManagement/directory/roleAssignments/$entity"}
+              "Parameters"                   = $args
+            }
+        )
+    }
+
+    Mock -CommandName Get-MgRoleManagementDirectoryRoleAssignment -MockWith $scriptblock -ModuleName Microsoft.Graph.Entra
+}
+
+Describe "Get-EntraDirectoryRoleAssignment" {
```
Comment (KenitoInc) [inline test/module/Entra/Get-EntraDirectoryRoleAssignment.Tests.ps1:32]: Add test for parameter `SearchString`

**Review by KenitoInc — APPROVED** [review]

**Pull request merged**